lib/Graph.dart

```diff
@@ -70,27 +72,36 @@ class Graph {
 
   void removeEdges(List<Edge> edges) => edges.forEach((it) => removeEdge(it));
 
-  void removeEdgeFromPredecessor(Node predecessor, Node current) {
-    _edges.removeWhere((edge) => edge.source == predecessor && edge.destination == current);
+  void removeEdgeFromPredecessor(Node? predecessor, Node? current) {
+    _edges.removeWhere(
+        (edge) => edge.source == predecessor && edge.destination == current);
   }
 
   bool hasNodes() => _nodes.isNotEmpty;
 
-  Edge getEdgeBetween(Node source, Node destination) =>
-      _edges.firstWhere((element) => element.source == source && element.destination == destination, orElse: ()=> null);
+  Edge? getEdgeBetween(Node source, Node? destination) =>
+      _edges.firstWhereOrNull((element) =>
+          element.source == source && element.destination == destination);
 
-  bool hasSuccessor(Node node) => _edges.any((element) => element.source == node);
+  bool hasSuccessor(Node? node) =>
```

Comment:
❓ here too.
`hasSuccessor(null)` == false
is not the same as `hasSuccessor(Node) == false`